### ISSUE TITLE
Web-components - Feature: Update Sass bundler to support Sass `2.0.0`

### ISSUE THREAD

**Issue by mahoneycm** [uswds/uswds-next#77]
# Description

The Sass legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.
[More info →](https://sass-lang.com/documentation/breaking-changes/legacy-js-api/)

## Proposed solution

Define sass API in Vite config

`vite.config.js`

```diff
export default defineConfig({
  css: {
    preprocessorOptions: {
      scss: {
+       api: "modern-compiler",
        includePaths: ["./node_modules/@uswds/uswds/packages"],
      },
    },
  },
```

## Additional information

In the `bundlers` section of the above guidance there is specific instructions for Vite.

> Vite still defaults to the legacy API, but you can similarly switch it by setting `api` to `"modern"` or `"modern-compiler"`. See [Vite’s documentation](https://vitejs.dev/config/shared-options.html#css-preprocessoroptions) for more detail
> 

>[!NOTE]
>~This work is currently blocked by uswds/uswds#6103~
>
>~Currently, switching to the modern api causes an error. I *think* that this is due to USWDS using the deprecated sass `render()` function.~

_This work should no longer be blocked thanks to uswds/uswds#6286_

**Comment by mejiaj** [comment]
@mahoneycm because we're blocked from updating here; do we have to mark uswds/uswds#6103 as a potentially breaking change and provide guidance?

**Comment by mahoneycm** [comment]
@mejiaj I don't believe so! By default, Vite is using the Legacy API and will continue to do so until we change it as stated in this issue here. 

The change that will be made in uswds/uswds#6103 seems only to affect some internal unit tests. I don't believe the users will be affected by this change at all.

**Comment by ethangardner** [comment]
This work is no longer necessary in this repo due to our [decision to use CSS](https://github.com/uswds/uswds-proposals/blob/main/decisions/0007-author-styles-in-css.md)
